scrypto/src/values.rs

```diff
@@ -136,6 +139,22 @@ impl ScryptoValue {
         node_ids
     }
 
+    pub fn global_references(&self) -> HashSet<GlobalAddress> {
+        let mut node_ids = HashSet::new();
+        for component_address in &self.refed_component_addresses {
+            node_ids.insert(GlobalAddress::Component(*component_address));
+        }
+        for resource_address in &self.resource_addresses {
+            node_ids.insert(GlobalAddress::Resource(*resource_address));
+        }
+        for non_fungible_address in &self.non_fungible_addresses {
+            node_ids.insert(GlobalAddress::Resource(
+                non_fungible_address.resource_address(),
+            ));
+        }
+        node_ids
```

Comment:
Why is `non_fungible_address` a global address? Do we intend direct call to non-fungible?

Reply:
It's not a global address, it's that we store a reference to the resource in the NonFungibleAddress so we must pass this around.

Reply:
Ah, another odd resource address inference.